src/main/java/com/epam/task/third/RegularTetrahedronListCreator.java

```diff
@@ -0,0 +1,100 @@
+package com.epam.task.third;
+
+import com.epam.task.third.data.DataException;
+import com.epam.task.third.data.DataReader;
+import com.epam.task.third.entity.Point;
+import com.epam.task.third.entity.Tetrahedron;
+import com.epam.task.third.parsing.TetrahedronPointsParser;
+import com.epam.task.third.validation.InputLineValidator;
+import com.epam.task.third.validation.RegularTetrahedronValidator;
+import com.epam.task.third.validation.Validator;
+
+import org.apache.logging.log4j.Level;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class RegularTetrahedronListCreator {
+
+    private DataReader reader;
+    private InputLineValidator lineValidator;
+    private TetrahedronPointsParser parser;
+    private static final Logger LOGGER = LogManager.getLogger();
+
+
+    public RegularTetrahedronListCreator(DataReader reader, InputLineValidator lineValidator,
+                                         TetrahedronPointsParser parser) {
+        this.reader = reader;
+        this.lineValidator = lineValidator;
+        this.parser = parser;
+    }
+
+    public List<Tetrahedron> createList(String filePath) {
+        List<Tetrahedron> tetrahedronList = new ArrayList<>();
+        try {
+            List<String> data = reader.readData(filePath);
+        LOGGER.log(Level.DEBUG,"Reading from resources - success");
+
+            for (String line : data) {
+
+                if (lineValidator.validate()) {
+                    TetrahedronPointsParser parser = new TetrahedronPointsParser();
+                    List<Point> tetrahedronData = parser.parse(line);
+                    Point a = tetrahedronData.get(0);
+                    Point b = tetrahedronData.get(1);
+                    Point c = tetrahedronData.get(2);
+                    Point d = tetrahedronData.get(3);
+
+                    RegularTetrahedronValidator regularityValidator =
```

Comment:
create Tetrahedron from line in a separate class, this class should be thin (its function is controller)